src/codemagic/apple/app_store_connect/provisioning/profiles.py

```diff
@@ -57,6 +57,21 @@ def create(self,
         """
         if profile_type.devices_not_allowed() and devices:
             raise ValueError(f'Cannot assign devices to profile with type {profile_type}')
+        elif profile_type.devices_required() and not devices:
+            if profile_type.is_tvos_profile:
+                device_type = 'tvOS'
+            elif profile_type.is_macos_profile:
+                device_type = 'macOS'
+            elif profile_type.is_ios_profile:
+                device_type = 'iOS'
+            else:
+                raise ValueError(f'Device type for profile type {profile_type} is unknown')
+
+            raise ValueError(
+                f'Cannot create profile: Apple requires that you register at least one {device_type} '
```

Comment:
Error message is misleading. We have no information here whatsoever to back up the claim that there are no devices registered in Apple Developer portal. What we know is that this profile creation request does not include any testing devices, while they are required for successful creation.

Reply:
Valid point, only concentrated on the case when we automatically try to create profiles (and therefore also fetch the device list ourselves). Will try to come up with something more covering yet still hint at a possible action.